spec/models/facility_spec.rb

```diff
@@ -126,4 +126,44 @@
       facility.cohort_analytics(:month, 3)
     end
   end
+
+  describe "Search" do
+    let(:facility_1) { create(:facility, name: "HWC Bahadurgarh") }
+    let(:facility_2) { create(:facility, name: "CHC Docomo", slug: "chc-docomo") }
+    let(:facility_3) { create(:facility, name: "CHC Porla Docomo") }
+
+    ["CHC", "DoCoMo", "DOCOMO", "docomo", "cHc", "chc"].each do |term|
+      it "returns results for case-insensitive searches: #{term.inspect}" do
+        expect(Facility.search_by_name(term)).to match_array(facility_2)
```

Comment:
Why do these searches not return facility 3?

Reply:
Great catch, my poor usage for `let` vs `let!` was the culprit.

Reply:
Aaaah gotcha gotcha. Neato, in that case this behaves the way I'd expect 👍

Reply:
I fixed the specs to be clearer and not leak `let` state or be confusing.